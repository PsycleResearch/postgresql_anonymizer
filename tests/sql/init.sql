BEGIN;

CREATE EXTENSION IF NOT EXISTS anon CASCADE;

SELECT anon.is_initialized() IS FALSE;

-- basic usage
SELECT anon.init();
SELECT anon.is_initialized();
SELECT anon.reset();

-- returns a WARNING and FALSE
SELECT anon.init('./does/not/exists/cd2ks3s/') IS FALSE;
SELECT anon.is_initialized() IS FALSE;

-- load alternate data dir
\! cp -pr data/default $PGDATA/tmp_anon_alternate_data
SELECT anon.init('tmp_anon_alternate_data');
\! rm -fr $PGDATA/tmp_anon_alternate_data
SELECT anon.reset();

-- load an empty data dir
-- returns a lot of NOTICE and FALSE
SELECT anon.init('pg_twophase') IS FALSE;
SELECT anon.is_initialized() IS FALSE;

-- backward compatibility with v0.6 and below
SELECT anon.load();

-- Returns a NOTICE and TRUE
SELECT anon.init();

SELECT anon.is_initialized() IS TRUE;

SELECT anon.reset();

SELECT anon.is_initialized() IS FALSE;

SELECT anon.start_dynamic_masking( autoload := FALSE );

SELECT anon.is_initialized() IS FALSE;

ROLLBACK;
