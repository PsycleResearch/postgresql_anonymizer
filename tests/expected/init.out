BEGIN;
CREATE EXTENSION IF NOT EXISTS anon CASCADE;
NOTICE:  installing required extension "tsm_system_rows"
NOTICE:  installing required extension "pgcrypto"
SELECT anon.is_initialized() IS FALSE;
 ?column? 
----------
 t
(1 row)

-- basic usage
SELECT anon.init();
 init 
------
 t
(1 row)

SELECT anon.is_initialized();
 is_initialized 
----------------
 t
(1 row)

SELECT anon.reset();
NOTICE:  truncate cascades to table "identifier"
 reset 
-------
 t
(1 row)

-- returns a WARNING and FALSE
SELECT anon.init('./does/not/exists/cd2ks3s/') IS FALSE;
NOTICE:  Data file './does/not/exists/cd2ks3s//identifiers_category.csv' is not present. Skipping.
NOTICE:  Data file './does/not/exists/cd2ks3s//identifier_fr_FR.csv' is not present. Skipping.
NOTICE:  Data file './does/not/exists/cd2ks3s//identifier_en_US.csv' is not present. Skipping.
NOTICE:  Data file './does/not/exists/cd2ks3s//city.csv' is not present. Skipping.
NOTICE:  Data file './does/not/exists/cd2ks3s//company.csv' is not present. Skipping.
NOTICE:  Data file './does/not/exists/cd2ks3s//email.csv' is not present. Skipping.
NOTICE:  Data file './does/not/exists/cd2ks3s//first_name.csv' is not present. Skipping.
NOTICE:  Data file './does/not/exists/cd2ks3s//iban.csv' is not present. Skipping.
NOTICE:  Data file './does/not/exists/cd2ks3s//last_name.csv' is not present. Skipping.
NOTICE:  Data file './does/not/exists/cd2ks3s//siret.csv' is not present. Skipping.
NOTICE:  Data file './does/not/exists/cd2ks3s//lorem_ipsum.csv' is not present. Skipping.
 ?column? 
----------
 t
(1 row)

SELECT anon.is_initialized() IS FALSE;
 ?column? 
----------
 t
(1 row)

-- load alternate data dir
\! cp -pr data/default /tmp/tmp_anon_alternate_data
SELECT anon.init('/tmp/tmp_anon_alternate_data');
